grid-client/subi/substrate_manager.go

```diff
@@ -149,7 +150,7 @@ func (s *SubstrateImpl) CancelContract(identity substrate.Identity, contractID u
 	s.m.Lock()
 	defer s.m.Unlock()
 
-	if err := s.Substrate.CancelContract(identity, contractID); err != nil && err.Error() != "ContractNotExists" {
+	if err := s.Substrate.CancelContract(identity, contractID); err != nil && !strings.Contains(err.Error(), "ContractNotExists") {
```

Comment:
isn't there a better way to check that error? like with Is or number constant comparison?

Reply:
with Is, we can't, because this error is not exposed from the tfchain client, it's only a string (https://github.com/threefoldtech/tfchain/blob/development/clients/tfchain-client-go/utils.go#L58)

Reply:
don't know what you mean with number constant comparison :(

Reply:
Alright, however I think it's the responsibility of the caller to check if the contract exists or not before deletion instead of introducing special handling in the code itself